pex/cli/commands/lock.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+from argparse import ArgumentParser, _ActionsContainer, _SubParsersAction
+
+from pex import resolver
+from pex.cli.command import BuildTimeCommand
+from pex.commands.command import Ok, OutputMixin, Result
+from pex.locked_resolve import LockConfiguration, LockStyle
+from pex.resolve import requirement_options, resolver_options, target_options
+from pex.variables import ENV
+
+
+class Lock(OutputMixin, BuildTimeCommand):
+    """Operate on PEX lock files."""
+
+    @staticmethod
+    def _add_resolve_options(parser):
+        # type: (_ActionsContainer) -> None
+        requirement_options.register(
+            parser.add_argument_group(
+                title="Requirement options",
+                description="Indicate which third party distributions should be resolved",
+            )
+        )
+        target_options.register(
+            parser.add_argument_group(
+                title="Target options",
+                description=(
+                    "Specify which interpreters and platforms resolved distributions must support."
+                ),
+            )
+        )
+        resolver_options.register(
+            parser.add_argument_group(
+                title="Resolver options",
+                description="Configure how third party distributions are resolved.",
+            ),
+            include_pex_repository=False,
+        )
+
+    @classmethod
+    def _add_create_arguments(cls, create_parser):
+        # type: (_ActionsContainer) -> None
+        create_parser.add_argument(
+            "--style",
+            default=LockStyle.STRICT,
+            choices=LockStyle.values(),
+            type=LockStyle.for_value,
+            help=(
+                "The style of lock to generate. The {strict!r} style is the default and generates "
+                "a lock file that contains exactly the distributions that would be used in a local "
+                "resolve. If an sdist would be used, the sdist is included, but if a wheel would "
+                "be used, an accompanying sdist will not be included. The {sources} style includes "
+                "locks containing wheels and the associated sdists when available.".format(
```

Comment:
Will `strict` include wheels for other platforms? (https://github.com/pantsbuild/pants/issues/12458)

I traced the code back to this and can't tell:

https://github.com/pantsbuild/pex/blob/0ab02beb91b406563b88049fd470beca82a12d7d/pex/pip.py#L472-L478

Reply:
It will not. This PRs issue, #1401, is all about platform specific locks. In order to get the Pip log data for other platforms, there are 4 Pip runtime patches that need to be applied and a new regex added here after that is done.

Reply:
So, I mean, yes - it will include wheels for other platforms when `--platform` is specified. I assumed you meant though for all possible other platforms, i.e.: totally platform agnostic. That's #1402.

Reply:
Yeah I did mean #1402. Cool, thanks for clarifying!